### PR TITLE
Feat/dv 79/add seeds to all entities

### PR DESCRIPTION
**Pull request by Eugene-pas**

### CODE REVIEW

File: Core/Validation/OfferCreateValidation.cs [hunk 1]
```diff
@@ -31,6 +34,18 @@ public OfferCreateValidation()
 
             RuleFor(offer => offer.Point.Order)
                 .GreaterThan(-1);
+
+            RuleFor(offer => offer.StartDate)
+                .NotEmpty()
+                .Must(date => date != DateTimeOffset.UtcNow)
```
Comment (marynakernychna) [inline Core/Validation/OfferCreateValidation.cs]: What if I want people to be able to respond to my offer right away?

File: Core/Validation/OfferCreateValidation.cs [hunk 2]
```diff
@@ -31,6 +34,18 @@ public OfferCreateValidation()
...
+                .WithMessage("StartDate cannot be created in the past tense");
```
Comment (marynakernychna) [inline Core/Validation/OfferCreateValidation.cs]: The word «tense» is misused here

File: Core/Validation/OfferCreateValidation.cs [hunk 3]
```diff
@@ -31,6 +34,18 @@ public OfferCreateValidation()
...
+
+            RuleFor(offer => (offer.ExpirationDate - offer.StartDate) < _hour)
```
Comment (marynakernychna) [inline Core/Validation/OfferCreateValidation.cs]: Use ready-made methods for working with dates

File: Infrastructure/Data/SeedData/SeedData.cs [hunk 4]
```diff
@@ -0,0 +1,936 @@
...
+                    Name = "User",
+                    NormalizedName = "USER",
+                    ConcurrencyStamp = USER_ROLE_ID
+                });
+
+        #endregion
+
+        #region SeedIdentityUserRole
+
+        public static void SeedIdentityUserRole(ModelBuilder builder) =>
```
Comment (marynakernychna) [inline Infrastructure/Data/SeedData/SeedData.cs:191]: It's all extra work

File: Core/Roles/OfferRoles.cs [hunk 5]
```diff
@@ -0,0 +1,8 @@
+﻿namespace Core.Roles
+{
+    public enum OfferRoles : int
+    {
+        SenderId = 1,
+        RecipientId = 2
+    }
+}
```
Comment (marynakernychna) [inline Core/Roles/OfferRoles.cs]: Empty line at EoF

File: Core/Services/OfferService.cs [hunk 6]
```diff
@@ -7,6 +7,7 @@
 using Core.Interfaces.CustomService;
 using Microsoft.AspNetCore.Identity;
 using System;
+using System.Net;
```
Comment (marynakernychna) [inline Core/Services/OfferService.cs]: Need to remove the extra using

File: Core/Services/OfferService.cs [hunk 7]
```diff
@@ -39,7 +40,7 @@ public class OfferService : IOfferService
         public async Task<OfferCreateDTO> CreateOfferAsync(OfferCreateDTO offerCreate, string userId)
         {
             ExceptionMethods.UserNullCheck(await _userManager.FindByIdAsync(userId));
-
+
```
Comment (marynakernychna) [inline Core/Services/OfferService.cs]: 3 tabs as a gift?

File: Core/Validation/OfferCreateValidation.cs [hunk 8]
```diff
@@ -1,10 +1,13 @@
-﻿using Core.DTO.OfferDTO;
+﻿using System;
+using System.ComponentModel;
```
Comment (marynakernychna) [inline Core/Validation/OfferCreateValidation.cs]: Need to remove the extra using

File: Infrastructure/Data/SeedData/SeedData.cs [hunk 9]
```diff
@@ -0,0 +1,936 @@
...
+        private static readonly string SERGEY_ID = Guid.NewGuid().ToString();
+        private static readonly string USER_ROLE_ID = Guid.NewGuid().ToString();
+
+        #endregion
+
+        private static readonly string[] Colors = new[] 
+            {"Червоний","Чорний","Білий","Сірий","Зелений","Синій","Жовтий"};
+        private static readonly PasswordHasher<User> PasswordHasher = new PasswordHasher<User>();
+        private static readonly Faker Faker = new Faker();
+        private static float _maxWeight = 1000f;
```
Comment (marynakernychna) [inline Infrastructure/Data/SeedData/SeedData.cs]: It is necessary to write names of variables in one style

File: Infrastructure/Data/SeedData/SeedData.cs [hunk 10]
```diff
@@ -0,0 +1,936 @@
...
+                    userAntonina,
+                    userSergey
+                    );
+        }
+
+        #endregion
+
+        #region SeedIdentityRole
+
+        public static void SeedIdentityRole(ModelBuilder builder) =>
```
Comment (marynakernychna) [inline Infrastructure/Data/SeedData/SeedData.cs:184]: You could add all the roles at once

File: Infrastructure/Data/SeedData/SeedData.cs [hunk 11]
```diff
@@ -0,0 +1,936 @@
...
+
+        #endregion
+
+        private static readonly string[] Colors = new[] 
+            {"Червоний","Чорний","Білий","Сірий","Зелений","Синій","Жовтий"};
+        private static readonly PasswordHasher<User> PasswordHasher = new PasswordHasher<User>();
+        private static readonly Faker Faker = new Faker();
+        private static float _maxWeight = 1000f;
+
+        public static void Seed(this ModelBuilder builder)
```
Comment (marynakernychna) [inline Infrastructure/Data/SeedData/SeedData.cs:46]: Why does the owner need so much data, including personal data of developers?
I think you should have created only roles, categories and one user (with random data, not real) by default

File: Infrastructure/Data/SeedData/SeedData.cs [hunk 12]
```diff
@@ -0,0 +1,957 @@
...
+        private static readonly string ROLE_ADMIN_ID = Guid.NewGuid().ToString();
+        private static readonly string ROLE_LOGIST_ID = Guid.NewGuid().ToString();
+
+        #endregion
+
+        private static readonly string[] Colors = new[] 
+            {"Червоний","Чорний","Білий","Сірий","Зелений","Синій","Жовтий"};
+        private static readonly PasswordHasher<User> PasswordHasher = new PasswordHasher<User>();
+        private static readonly Faker Faker = new Faker();
+        private static readonly float MaxWeight = 1000f;
```
Comment (marynakernychna) [inline Infrastructure/Data/SeedData/SeedData.cs]: We call the variables in the project not in this style

File: Infrastructure/Data/SeedData/SeedData.cs [hunk 13]
```diff
@@ -0,0 +1,957 @@
...
+
+        #endregion
+
+        #region SeedCarCategory
+
+        public static void SeedCarCategory(ModelBuilder builder) =>
+            builder.Entity<CarCategory>().HasData(
+                new CarCategory()
+                {
+                    Id = (int)CarCategories.A,
```
Comment (marynakernychna) [inline Infrastructure/Data/SeedData/SeedData.cs]: Why can't we just write an id by number like in the others Seeds?
How do I know from the name of this constant that I can get an id from there?

File: Infrastructure/Data/SeedData/SeedData.cs [hunk 14]
```diff
@@ -0,0 +1,957 @@
...
+                    Id = 1,
+                    Description = $"{Faker.Random.Words(Faker.Random.Int(10, 30))}.",
+                    IsClosed = false,
+                    GoodsWeight = Faker.Random.Float(0.1f, MaxWeight),
+                    StartDate = DateTimeOffset.Parse($"{Faker.Random.Int(1, 5)}.07.2022 {Faker.Random.Int(5, 12)}:00"),
+                    ExpirationDate = DateTimeOffset.Parse($"{Faker.Random.Int(5, 10)}.07.2022 {Faker.Random.Int(12, 23)}:00"),
+                    GoodCategoryId = Faker.Random.Int(1, 7),
+                    OfferPointId = 1,
+                    OfferCreatorId = EUGEN_ID,
+                    CreatorRoleId = (int)OfferRoles.Sender
```
Comment (marynakernychna) [inline Infrastructure/Data/SeedData/SeedData.cs]: How do I know from the name of this constant that I can get an id from there?

File: Core/Services/OfferService.cs [hunk 15]
```diff
@@ -39,7 +39,6 @@ public class OfferService : IOfferService
         public async Task<OfferCreateDTO> CreateOfferAsync(OfferCreateDTO offerCreate, string userId)
         {
             ExceptionMethods.UserNullCheck(await _userManager.FindByIdAsync(userId));
-
```
Comment (marynakernychna) [inline Core/Services/OfferService.cs]: Why did you delete the empty line?

File: Infrastructure/Data/SeedData/SeedData.cs [hunk 16]
```diff
@@ -0,0 +1,957 @@
...
+
+        #endregion
+
+        #region SeedRole
+
+        public static void SeedRole(ModelBuilder builder) =>
+            builder.Entity<Role>().HasData(
+                new Role()
+                {
+                    Id = (int)OfferRoles.Sender,
```
Comment (marynakernychna) [inline Infrastructure/Data/SeedData/SeedData.cs:256]: Why can't we just write an id by number like in the others Seeds

File: Core/Constants/OfferRoles.cs [hunk 17]
```diff
@@ -0,0 +1,8 @@
+﻿namespace Core.Constants
+{
+    public enum OfferRoles : int
```
Comment (marynakernychna) [inline Core/Constants/OfferRoles.cs:3]: Why do we save the id?

File: Core/Constants/IdentityRoles.cs [hunk 18]
```diff
@@ -0,0 +1,9 @@
+﻿namespace Core.Constants
+{
+    public enum IdentityRoleNames
```
Comment (marynakernychna) [inline Core/Constants/IdentityRoles.cs]: Why don't we create string constants instead of enum so that we don't convert them to a string later?

**Review by marynakernychna — APPROVED** [review]

File: Core/Validation/OfferCreateValidation.cs [hunk 19]
```diff
@@ -31,6 +33,20 @@ public OfferCreateValidation()
...
+            RuleFor(offer => offer.StartDate)
+                .NotEmpty()
+                .Must(date => date >= DateTimeOffset.UtcNow)
+                .WithMessage("StartDate cannot be created in the past");
+
+            RuleFor(offer => offer.ExpirationDate)
+                .NotEmpty();
+
+            RuleFor(offer => offer.ExpirationDate.Subtract(offer.StartDate) < _hour)
+                .Must(date => date == false)
```
Comment (Grasmus) [inline Core/Validation/OfferCreateValidation.cs]: Why do we compare bool variable with false operator?

**Review by Grasmus — APPROVED** [review]
